### PR TITLE
Adding a getconfig command for Wazuh-DB

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#11732|

## Description

This PR adds a new Wazuh-DB command to get the current configuration, from `internal_options` or from `ossec.conf`.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)

- [x] Configuration on demand reports new parameters

### CODE REVIEW

**Comment by pereyra-m** [comment]
### Scan-build

The **scan-build** tool reports 3 issues that were solved in https://github.com/wazuh/wazuh/pull/11739
scan_build



### Manual tests

The **WDB_tester.py** script was used to test this command. All the test cases are in the **test_wdb.yaml** file ([wdb_getconfig_tests.zip](https://github.com/wazuh/wazuh/files/7831555/wdb_getconfig_tests.zip)).
These are the results

testing_wdb_get_config

File: src/wazuh_db/wdb.h [hunk 1]
```diff
@@ -2365,4 +2365,26 @@ int wdb_task_get_upgrade_task_by_agent_id(wdb_t* wdb, int agent_id, char **node,
 // Finalize a statement securely
 #define wdb_finalize(x) { if (x) { sqlite3_finalize(x); x = NULL; } }
 
+/**
+ * @brief Method to parse the "wazuhdb getconfig" commands.
+ *
+ * @param config_source Where the config will be read from: "internal" or "wdb"
```
Comment (palaciosjeremias) [inline src/wazuh_db/wdb.h:2371]: Should this be "wdb" or i.e. "wdb section"

**Review by palaciosjeremias — APPROVED** [review]
LGTM!

File: src/wazuh_db/wdb.h (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb.h:2371]: Done!

File: src/config/wazuh_db-config.c [hunk 2]
```diff
@@ -72,18 +72,18 @@ int Read_WazuhDB_Backup(const OS_XML *xml, xml_node * node, int const BACKUP_NOD
     for (int i = 0; chld_node[i]; i++) {
         if (!chld_node[i]->element) {
             merror(XML_ELEMNULL);
-            os_free(chld_node);
+            OS_ClearNode(chld_node);
```
Comment (DProvinciani) [inline src/config/wazuh_db-config.c:75]: Looks better and more consistent using `OS_ClearNode`. Good improvement.

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**